File: ThaiTuanERP2025.Server/ThaiTuanERP2025.Application/Account/Dtos/UserDto.cs
namespace ThaiTuanERP2025.Application.Account.Dtos
{
	public class UserDto
	{
		public Guid Id { get; set; }
		public string FullName { get; set; } = default!;
		public string Username { get; set; } = default!;
		public string EmployeeCode { get; set; } = default!;
		public string Position { get; set; } = default!;
		public string? Email { get; set; }
		public string? Phone { get; set; }

		public string? AvatarFileId { get; set; }
		public string? AvatarFileObjectKey { get; set; }

		public List<RoleDto> Roles { get; set; } = new();
		public List<PermissionDto> Permissions { get; set; } = new();

		public Guid? DepartmentId { get; set; }

		public DepartmentDto? Department { get; set; } = default!;
	}

	
}

File: ThaiTuanERP2025.Client/src/app/pages/account/dtos/user.dto.ts
import { DepartmentDto } from "./department.dto";
import { UserRole } from "./user-roles.enum";

export interface UserDto {
      id?: string;
      fullName: string;
      username: string;
      employeeCode: string;
      email?: string;
      password: string;
      avatarUrl?: string;
      role: UserRole;
      phone?: string;
      departmentId?: string;  
      department?: DepartmentDto;
      position: string;
}

export interface CreateUserDto {
      fullName: string;
      username: string;
      employeeCode: string;
      email?: string;
      password: string;
      role: string;
      phone?: string;
      departmentId?: string;
      position: string;
}